### PR TITLE
3 body Jastrow forces

### PR DESCRIPTION
**Pull request by rcclay**
Please review the [developer documentation](https://github.com/QMCPACK/qmcpack/wiki/Development-workflow)
on the wiki of this project that contains help and requirements.

## Proposed changes
This adds finite difference calculation of the wave function ion gradients required to use 3 body Jastrow factors with ZVZB force calculations.  This will remain until analytic derivatives are derived and implemented.  
  
## What type(s) of changes does this code introduce?

- New feature

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
Intel Xeon Haswell
## Checklist

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

NOTE:  There isn't a unit test for this yet, but WaveFunctionTester can and has been used to validate this feature.

### CODE REVIEW

**Comment by rcclay** [comment]
Looks like I was sloppy with MP types.  I'll fix tomorrow.

File: src/QMCWaveFunctions/Jastrow/JeeIOrbitalSoA.h [hunk 1]
```diff
@@ -969,6 +986,106 @@ class JeeIOrbitalSoA : public WaveFunctionComponent
...
+      P.update();
+      LogValueType log_m = evaluateLog(P, Gm, Lm);
+      QTFull::RealType log_p_r(0.0), log_m_r(0.0);
+      
+      log_p_r=log_p.real();
+      log_m_r=log_m.real();
+      dG=Gp-Gm;
+      dL=Lp-Lm;
+      //symmetric finite difference formula for gradient.
+      g_return[iondim] = c1 * (log_p_r - log_m_r);
```
Comment (ye-luo) [inline src/QMCWaveFunctions/Jastrow/JeeIOrbitalSoA.h:1077]: This formulation is only correct without node-crossing. I noticed this pattern in wftest and we need certain protection there as well.

Comment (ye-luo) [reply src/QMCWaveFunctions/Jastrow/JeeIOrbitalSoA.h:1077]: OK. It is fine here since Jastrow is bosonic. But not a general case in wftest.

**Comment by ye-luo** [comment]
Test this please.

File: src/QMCWaveFunctions/Jastrow/JeeIOrbitalSoA.h (hunk 1 above)
Comment (rcclay) [reply src/QMCWaveFunctions/Jastrow/JeeIOrbitalSoA.h:1077]: What are you thinking?  Check for a sign difference and abort if that happens?  Abort if log_p or log_m gets really really negative?

Comment (rcclay) [reply src/QMCWaveFunctions/Jastrow/JeeIOrbitalSoA.h:1077]: It's moot for the Jastrow, but a valid concern for wftester that I hadn't considered.

Comment (ye-luo) [reply src/QMCWaveFunctions/Jastrow/JeeIOrbitalSoA.h:1077]: No problem here in the Jastrow.
In the generic case, the sign of psi matters and so as the phase of log(psi) not the sign of log(psi).

Comment (rcclay) [reply src/QMCWaveFunctions/Jastrow/JeeIOrbitalSoA.h:1077]: Gotcha.

**Pull request merged**